utils_linux.go

```diff
@@ -248,11 +249,19 @@ func (r *runner) run(config *specs.Process) (int, error) {
 		}
 		process.ExtraFiles = append(process.ExtraFiles, os.NewFile(uintptr(i), "PreserveFD:"+strconv.Itoa(i)))
 	}
-	detach := r.detach || (r.action == CT_ACT_CREATE)
 	// Setting up IO is a two stage process. We need to modify process to deal
 	// with detaching containers, and then we get a tty after the container has
 	// started.
-	handlerCh := newSignalHandler(r.enableSubreaper, r.notifySocket)
+	if r.enableSubreaper {
+		// set us as the subreaper before registering the signal handler for the container
```

Comment:
I guess it is fine to do this even on !detached containers, that we don't register the signal handler?

Reply:
(You probably meant to ask "even for detached containers")

Hmm, indeed it does not make sense to register ourselves as the child reaper if we're not going to do it, and we are only doing it in `handler.forward()`.

Reply:
Yes, I think all cleanup actions should be in a defer function, it will be more clear.
https://github.com/opencontainers/runc/pull/4661/files#r2004617978

Reply:
> Hmm, indeed it does not make sense to register ourselves as the child reaper if we're not going to do it, and we are only doing it in `handler.forward()`.

I want to keep it as is, because it's the same as before.

Reply:
Okay :)